game-app/game-core/src/main/java/org/triplea/util/Awt.java

```diff
@@ -0,0 +1,36 @@
+package org.triplea.util;
+
+import java.awt.Graphics2D;
+import java.awt.Image;
+import java.awt.image.BufferedImage;
+import lombok.experimental.UtilityClass;
+
+/**
+ * container for AWT related helper functions
+ */
+
+@UtilityClass
+public class Awt {
+  /**
+   * @param image to return a buffered image of
+   * @param requiredType type of the BufferedImage to return
+   * @return a BufferedImage
+   */
+  public BufferedImage getBufferedImage(final Image image, final int requiredType) {
```

Comment:
BufferedImage is using "evil" magic ints as status flags. EG: `BufferedImage.TYPE_INT_ARGB`

When first encountering this method, I wondered what 'requiredType' meant. Would this be a good candidate to replace with an enum? Having an enum map the allowed values into the method to the magic types would make the API type-safe and far more self documenting. As is, I'd have trouble using this method by just looking at any of the: API, code, and javadocs. I'm left wondering which values of BufferedImage can exist, where to get those, and why I would want to use differing options. I think having an enum would help solve almost all of those problems.

Reply:
I agree with you regarding "evil INTs". If Awt was my library, I probably wouldn't habe introduced them. Awt being as it is, I used it they way it was designed. I like the idea to hold up good coding standards - e.g. using enums instead if INTs for image types. That would require mappers between TripleA-Quality and used library quality. This would add an additional level of complexity though (and additional lines of code). And it would assume that the respective programmer is more familiar with the TripleA code than with the library - which in case of Awt might be a stretched assumption.

Reply:
I ahve improved the javadoc in the latest commit